### PR TITLE
feat: file available offline

### PR DESCRIPTION
**Pull request by enguerran**
Let user to set a file to be available on offline:
- [x] make a file as offline
- [x] see which files are available offline
- [x] open a offline file even on offline mode
- [x] prettify UI (see screenshot)
- __one more thing:__ refactor configureStore

### CODE REVIEW

**Comment by enguerran** [comment]
[(3) [FIL]📱 Rendre disponible un fichier hors-ligne (one-shot)](https://trello.com/c/uoSCGOws/442-3-fil%F0%9F%93%B1-rendre-disponible-un-fichier-hors-ligne-one-shot)

**Comment by enguerran** [comment]
Thanks @GoOz, it's very better

localhost-8084- galaxy s5

File: src/drive/styles/table.styl [hunk 1]
```diff
@@ -131,6 +138,10 @@ mime-types('fil-file-', '../assets/icons/')
     background-size cover
     background-position center center
 
+// .fil-available-offline
```
Comment (y-lohse) [inline src/drive/styles/table.styl]: Can we get rid of this commented code?

Comment (GoOz) [reply src/drive/styles/table.styl]: ah my bad…

File: src/drive/ducks/files/Container.jsx [hunk 2]
```diff
@@ -59,6 +62,14 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
         action: selected => dispatch(startRenamingAsync(selected[0])),
         displayCondition: (selections) => selections.length === 1 && isFile(selections[0])
       },
+      availableOffline: {
+        action: selected => dispatch(toggleAvailableOffline(selected[0])),
+        displayCondition: (selections) => __TARGET__ === 'mobile' && selections.length === 1 && isFile(selections[0]),
+        checkbox: {
+          value: true,
+          onChange: console.log.bind(console)
```
[the remark this answers is not in the document]
Comment (enguerran) [reply src/drive/ducks/files/Container.jsx]: As I have no perspective at all of such an additional behavior in action menu, I cannot decide if we want only one action to perform.
In my own case, there is no need to keep an `onChange` on the toggle as the toggle is strongly connected to the action. When we click on the menu item, it toggles the value and the toggle component is rendered in the same way.
Even the value is not necessary here as we `connect` the MenuItem to the specific slice of the global state.

Then I will review this. Thank you for your hindsight.

File: src/drive/components/FileActionMenu.jsx [hunk 3]
```diff
@@ -25,19 +27,26 @@ class MenuItem extends Component {
...
+          {checkbox && <Toggle id={children} checked={checkbox.value} onToggle={checkbox.onChange} />}
         </a>
-      </Item>
+      </div>
     )
   }
 }
 
+const mapStateToProps = (state, ownProps) => ({
+  checkbox: ownProps.checkbox && { ...ownProps.checkbox, value: isAvailableOffline(state)(ownProps.files[0].id) }
```
[the remark this answers is not in the document]
Comment (enguerran) [reply src/drive/components/FileActionMenu.jsx]: The only other place is in the `files/Container.jsx` (which is very badly named, imho).
But, there is no access to the global state, except the third parameter of [the `connect` function](https://github.com/cozy/cozy-drive/pull/351/files#diff-791ebcf61511f68da1d94d3b8f95ff19R95).

In the other hand, redux and its beloved architecture give us the ability to stay generic and stateless for the great majority of our application's components. Id est, if we want to flow data through components to a, lets say, `MenuItem`, we have to connect the `MenuItem` and we must avoid data flow through stateless and irresponsible components between a parent component and its children.

---

Le seul autre endroit c'est dans `files/Container.jsx` dans lequel il n'y a pas d'autre moyen d'accéder au «state» que via le 3e paramètre de la fonction `connect`.

D'un autre côté, redux et son architecture bien-aimée nous permette de rester le plus générique possible en utilisant des composants stateless. C'est à dire que si nous voulons faire descendre des données d'un composant à un autre, au lieu d'un gros data flow par des props, on peut connecter l'élément qui doit être smart.

[the remark this answers is not in the document]
Comment (goldoraf) [reply src/drive/components/FileActionMenu.jsx]: I'm OK for connecting a menu item, it makes sense to connect more at the leaf level and less at the branch level, but I'm uneasy with connecting all menu items just for one item. More on this in the next comment.

File: src/drive/ducks/files/Container.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (goldoraf) [reply src/drive/ducks/files/Container.jsx]: I don't like this `checkbox` option to be honest. We tried to have a generic definition of actions to be included in the action menu, and this new option is a little bit too specific and hard to understand to my taste. What do you think of the idea of having a `component` option instead? Something like this:
```jsx
availableOffline: {
  action: selected => dispatch(toggleAvailableOffline(selected[0])),
  displayCondition: (selections) => __TARGET__ === 'mobile' && selections.length === 1 && isFile(selections[0]),
  component: <ConnectedToggleMenuItem />
}
```
It'd allow us to use a specific component for *some* menu items without sacrificing too much genericity, and only the necessary components would get connected to the redux store.
What do you think?

**Comment by enguerran** [comment]
@y-lohse I let you a chance to review changes on FileActionMenu. If you think it's okay, I'll merge this PR.

**Pull request merged**